test/poll_for_state.go

```diff
@@ -0,0 +1,56 @@
+package test
+
+import (
+	"context"
+	"errors"
+
+	"github.com/strangelove-ventures/ibctest/ibc"
+)
+
+// ChainAcker is a chain that can get its acknowledgements at a specified height
+type ChainAcker interface {
+	ChainHeighter
+	Acknowledgements(ctx context.Context, height uint64) ([]ibc.PacketAcknowledgement, error)
+}
+
+// PollForAck attempts to find an acknowledgement containing a packet equal to the packet argument.
+// Polling starts at startHeight and continues until maxHeight. It is safe to call this function even if
+// the chain has yet to produce blocks for the target min/max height range. Polling delays until heights exist
+// on the chain. If no acknowledgement found, returns a not-found error.
```

Comment:
This may make the case for an explicit `ErrNotFound` or similar. My inclination is that you only need to know if you found acks or not. The reason why is less important. Adding a 3rd bool return argument is unecessarily complex for this use case imo.